code/modules/mob/living/carbon/xenomorph/castes/hivemind/hivemind.dm

```diff
@@ -72,7 +151,26 @@
 /mob/living/carbon/xenomorph/hivemind/proc/return_to_core()
 	forceMove(get_turf(core))
 
+///Start the teleportation process to send the hivemind manifestation to the selected turf
+/mob/living/carbon/xenomorph/hivemind/proc/start_teleport(turf/T)
+	flick("Hivemind_materialisation_reverse", src)
+	can_move = FALSE
+	addtimer(CALLBACK(src, .proc/end_teleport, T), TIME_TO_TRANSFORM)
```

Comment:
why not a do after?

Reply:
Because a do after is interruptible

Reply:
whats the trouble of it being interruptible, it setting can_move to false and adding a timer is wtf

Reply:
because i want players to commit to a teleport. Also, this would fuck up the animation greatly.

I don't want players to be able to move at all while teleporting, a do_after doesn't do the same thing